src/OutputReaders/set_field_time_series.jl

```diff
@@ -7,23 +7,34 @@ using Oceananigans.Architectures: cpu_architecture
 
 iterations_from_file(file) = parse.(Int, keys(file["timeseries/t"]))
 
-find_time_index(time::Number, file_times)       = findfirst(t -> t ≈ time, file_times)
-find_time_index(time::AbstractTime, file_times) = findfirst(t -> t == time, file_times)
+function find_time_index(time::Number, file_times, dt)
+    ϵ = 100 * eps(eltype(file_times))
+    return findfirst(t -> isapprox(t, time, atol=ϵ*dt), file_times)
+end
+
+find_time_index(time::AbstractTime, file_times, dt) = findfirst(t -> t == time, file_times)
 
 function set!(fts::InMemoryFTS, path::String=fts.path, name::String=fts.name; warn_missing_data=true)
     file = jldopen(path; fts.reader_kw...)
     file_iterations = iterations_from_file(file)
     file_times = [file["timeseries/t/$i"] for i in file_iterations]
     close(file)
-
+    
+    # Compute a timescale for comparisons
+    dt = mean(file_times[2:end] - file_times[1:end-1])
```

Comment:
why don't we compute this within `find_time_index`? just a thought...
or something like:

```julia
function find_time_index(time::Number, file_times; time_scale = mean(diff(file_times)))
    ϵ = 100 * eps(eltype(file_times))
    return findfirst(t -> isapprox(t, time, atol=ϵ*time_scale), file_times)
end
```

Reply:
Precomputing serves two purposes: 1) we don't recompute, which could become important for very long time series, and 2) as a point of design, this allows us to change the estimate of the characteristic time increment later. Right now we use the mean of all time increments, but this is not the only choice one might make.